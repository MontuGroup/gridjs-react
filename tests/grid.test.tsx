import * as React from "react";
import { mount } from "enzyme";
import Grid from "../src/grid";

const flushPromises = () => new Promise(setImmediate);

describe("Grid component", () => {
  it("should render a table without header", async () => {
    const table = mount<Grid>(<Grid data={[[1, 2, 3]]} />);
    await flushPromises();
    table.update();
    expect(table.html()).toMatchSnapshot();
  });

  it("should render a table with headers", async () => {
    const table = mount<Grid>(
      <Grid data={[[1, 2, 3]]} columns={["a", "b", "c"]} />
    );

    await flushPromises();
    table.update();
    expect(table.html()).toMatchSnapshot();
  });

  it("should render a table with width", async () => {
    const table = mount<Grid>(
      <Grid data={[[1, 2, 3]]} columns={["a", "b", "c"]} width={"500px"} />
    );

    await flushPromises();
    table.update();
    expect(table.html()).toMatchSnapshot();
  });

  it("should render a table with search", async () => {
    const table = mount<Grid>(
      <Grid data={[[1, 2, 3]]} columns={["a", "b", "c"]} search={true} />
    );

    await flushPromises();
    table.update();
    expect(table.html()).toMatchSnapshot();
  });

  it("should render a table with pagination and search", async () => {
    const table = mount<Grid>(
      <Grid
        data={[
          [1, 2, 3],
          [4, 5, 6],
          [7, 8, 9],
        ]}
        columns={["a", "b", "c"]}
        search={true}
        pagination={{
          enabled: true,
          limit: 1,
        }}
      />
    );

    await flushPromises();
    table.update();
    expect(table.html()).toMatchSnapshot();
  });
});
