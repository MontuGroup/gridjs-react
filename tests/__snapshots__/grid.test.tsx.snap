// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid component should render a table with headers 1`] = `"<div><div role=\\"complementary\\" class=\\"gridjs gridjs-container\\" style=\\"width: 100%;\\"><div class=\\"gridjs-wrapper\\" style=\\"width: 100%; height: auto;\\"><table role=\\"grid\\" class=\\"gridjs-table\\" style=\\"width: 100%; height: auto;\\"><thead class=\\"gridjs-thead\\"><tr class=\\"gridjs-tr\\"><th data-column-id=\\"a\\" class=\\"gridjs-th\\">a</th><th data-column-id=\\"b\\" class=\\"gridjs-th\\">b</th><th data-column-id=\\"c\\" class=\\"gridjs-th\\">c</th></tr></thead><tbody class=\\"gridjs-tbody\\"><tr class=\\"gridjs-tr\\"><td data-column-id=\\"a\\" class=\\"gridjs-td\\">1</td><td data-column-id=\\"b\\" class=\\"gridjs-td\\">2</td><td data-column-id=\\"c\\" class=\\"gridjs-td\\">3</td></tr></tbody></table></div></div><div id=\\"gridjs-temp\\" class=\\"gridjs-temp\\"></div></div>"`;

exports[`Grid component should render a table with pagination and search 1`] = `"<div><div role=\\"complementary\\" class=\\"gridjs gridjs-container\\" style=\\"width: 100%;\\"><div class=\\"gridjs-head\\"><div class=\\"gridjs-search\\"><input type=\\"search\\" placeholder=\\"Type a keyword...\\" aria-label=\\"Type a keyword...\\" class=\\"gridjs-input gridjs-search-input\\"></div></div><div class=\\"gridjs-wrapper\\" style=\\"width: 100%; height: auto;\\"><table role=\\"grid\\" class=\\"gridjs-table\\" style=\\"width: 100%; height: auto;\\"><thead class=\\"gridjs-thead\\"><tr class=\\"gridjs-tr\\"><th data-column-id=\\"a\\" class=\\"gridjs-th\\">a</th><th data-column-id=\\"b\\" class=\\"gridjs-th\\">b</th><th data-column-id=\\"c\\" class=\\"gridjs-th\\">c</th></tr></thead><tbody class=\\"gridjs-tbody\\"><tr class=\\"gridjs-tr\\"><td data-column-id=\\"a\\" class=\\"gridjs-td\\">1</td><td data-column-id=\\"b\\" class=\\"gridjs-td\\">2</td><td data-column-id=\\"c\\" class=\\"gridjs-td\\">3</td></tr></tbody></table></div><div class=\\"gridjs-footer\\"><div class=\\"gridjs-pagination\\"><div role=\\"status\\" class=\\"gridjs-summary\\" title=\\"Page 1 of 3\\">Showing <b>1</b> to <b>1</b> of <b>3</b> results</div><div class=\\"gridjs-pages\\"><button tabindex=\\"0\\" disabled=\\"\\">Previous</button><button tabindex=\\"0\\" class=\\"gridjs-currentPage\\" title=\\"Page 1\\">1</button><button tabindex=\\"0\\" class=\\"\\" title=\\"Page 2\\">2</button><button tabindex=\\"0\\" class=\\"\\" title=\\"Page 3\\">3</button><button tabindex=\\"0\\">Next</button></div></div></div></div><div id=\\"gridjs-temp\\" class=\\"gridjs-temp\\"></div></div>"`;

exports[`Grid component should render a table with search 1`] = `"<div><div role=\\"complementary\\" class=\\"gridjs gridjs-container\\" style=\\"width: 100%;\\"><div class=\\"gridjs-head\\"><div class=\\"gridjs-search\\"><input type=\\"search\\" placeholder=\\"Type a keyword...\\" aria-label=\\"Type a keyword...\\" class=\\"gridjs-input gridjs-search-input\\"></div></div><div class=\\"gridjs-wrapper\\" style=\\"width: 100%; height: auto;\\"><table role=\\"grid\\" class=\\"gridjs-table\\" style=\\"width: 100%; height: auto;\\"><thead class=\\"gridjs-thead\\"><tr class=\\"gridjs-tr\\"><th data-column-id=\\"a\\" class=\\"gridjs-th\\">a</th><th data-column-id=\\"b\\" class=\\"gridjs-th\\">b</th><th data-column-id=\\"c\\" class=\\"gridjs-th\\">c</th></tr></thead><tbody class=\\"gridjs-tbody\\"><tr class=\\"gridjs-tr\\"><td data-column-id=\\"a\\" class=\\"gridjs-td\\">1</td><td data-column-id=\\"b\\" class=\\"gridjs-td\\">2</td><td data-column-id=\\"c\\" class=\\"gridjs-td\\">3</td></tr></tbody></table></div></div><div id=\\"gridjs-temp\\" class=\\"gridjs-temp\\"></div></div>"`;

exports[`Grid component should render a table with the ReactWrapper class 1`] = `"<div><div role=\\"complementary\\" class=\\"gridjs gridjs-container\\" style=\\"width: 100%;\\"><div class=\\"gridjs-wrapper\\" style=\\"width: 100%; height: auto;\\"><table role=\\"grid\\" class=\\"gridjs-table\\" style=\\"width: 100%; height: auto;\\"><thead class=\\"gridjs-thead\\"><tr class=\\"gridjs-tr\\"><th data-column-id=\\"a\\" class=\\"gridjs-th\\">a</th><th data-column-id=\\"b\\" class=\\"gridjs-th\\">b</th><th data-column-id=\\"c\\" class=\\"gridjs-th\\">c</th></tr></thead><tbody class=\\"gridjs-tbody\\"><tr class=\\"gridjs-tr\\"><td data-column-id=\\"a\\" class=\\"gridjs-td\\">1</td><td data-column-id=\\"b\\" class=\\"gridjs-td\\">2</td><td data-column-id=\\"c\\" class=\\"gridjs-td\\">3</td></tr><tr class=\\"gridjs-tr\\"><td data-column-id=\\"a\\" class=\\"gridjs-td\\">4</td><td data-column-id=\\"b\\" class=\\"gridjs-td\\">5</td><td data-column-id=\\"c\\" class=\\"gridjs-td\\">6</td></tr><tr class=\\"gridjs-tr\\"><td data-column-id=\\"a\\" class=\\"gridjs-td\\"><div><b>7</b></div></td><td data-column-id=\\"b\\" class=\\"gridjs-td\\"><div><b>8</b></div></td><td data-column-id=\\"c\\" class=\\"gridjs-td\\"><span><b>8</b></span></td></tr></tbody></table></div></div><div id=\\"gridjs-temp\\" class=\\"gridjs-temp\\"></div></div>"`;

exports[`Grid component should render a table with the wrapper function 1`] = `"<div><div role=\\"complementary\\" class=\\"gridjs gridjs-container\\" style=\\"width: 100%;\\"><div class=\\"gridjs-wrapper\\" style=\\"width: 100%; height: auto;\\"><table role=\\"grid\\" class=\\"gridjs-table\\" style=\\"width: 100%; height: auto;\\"><thead class=\\"gridjs-thead\\"><tr class=\\"gridjs-tr\\"><th data-column-id=\\"a\\" class=\\"gridjs-th\\">a</th><th data-column-id=\\"b\\" class=\\"gridjs-th\\">b</th><th data-column-id=\\"c\\" class=\\"gridjs-th\\">c</th></tr></thead><tbody class=\\"gridjs-tbody\\"><tr class=\\"gridjs-tr\\"><td data-column-id=\\"a\\" class=\\"gridjs-td\\">1</td><td data-column-id=\\"b\\" class=\\"gridjs-td\\">2</td><td data-column-id=\\"c\\" class=\\"gridjs-td\\">3</td></tr><tr class=\\"gridjs-tr\\"><td data-column-id=\\"a\\" class=\\"gridjs-td\\">4</td><td data-column-id=\\"b\\" class=\\"gridjs-td\\">5</td><td data-column-id=\\"c\\" class=\\"gridjs-td\\">6</td></tr><tr class=\\"gridjs-tr\\"><td data-column-id=\\"a\\" class=\\"gridjs-td\\"><div><b>7</b></div></td><td data-column-id=\\"b\\" class=\\"gridjs-td\\"><div><b>8</b></div></td><td data-column-id=\\"c\\" class=\\"gridjs-td\\"><span><span>9</span></span></td></tr></tbody></table></div></div><div id=\\"gridjs-temp\\" class=\\"gridjs-temp\\"></div></div>"`;

exports[`Grid component should render a table with width 1`] = `"<div><div role=\\"complementary\\" class=\\"gridjs gridjs-container\\" style=\\"width: 500px;\\"><div class=\\"gridjs-wrapper\\" style=\\"width: 500px; height: auto;\\"><table role=\\"grid\\" class=\\"gridjs-table\\" style=\\"width: 500px; height: auto;\\"><thead class=\\"gridjs-thead\\"><tr class=\\"gridjs-tr\\"><th data-column-id=\\"a\\" class=\\"gridjs-th\\">a</th><th data-column-id=\\"b\\" class=\\"gridjs-th\\">b</th><th data-column-id=\\"c\\" class=\\"gridjs-th\\">c</th></tr></thead><tbody class=\\"gridjs-tbody\\"><tr class=\\"gridjs-tr\\"><td data-column-id=\\"a\\" class=\\"gridjs-td\\">1</td><td data-column-id=\\"b\\" class=\\"gridjs-td\\">2</td><td data-column-id=\\"c\\" class=\\"gridjs-td\\">3</td></tr></tbody></table></div></div><div id=\\"gridjs-temp\\" class=\\"gridjs-temp\\"></div></div>"`;

exports[`Grid component should render a table without header 1`] = `"<div><div role=\\"complementary\\" class=\\"gridjs gridjs-container\\" style=\\"width: 100%;\\"><div class=\\"gridjs-wrapper\\" style=\\"width: 100%; height: auto;\\"><table role=\\"grid\\" class=\\"gridjs-table\\" style=\\"width: 100%; height: auto;\\"><tbody class=\\"gridjs-tbody\\"><tr class=\\"gridjs-tr\\"><td class=\\"gridjs-td\\">1</td><td class=\\"gridjs-td\\">2</td><td class=\\"gridjs-td\\">3</td></tr></tbody></table></div></div><div id=\\"gridjs-temp\\" class=\\"gridjs-temp\\"></div></div>"`;
